packages/gallery/src/components/gallery/presets/propTypes.js

```diff
@@ -0,0 +1,76 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import consts from '../../../common/constants';
+
+export const basePropTypes = {
+  domId: PropTypes.string.isRequired,
+  items: PropTypes.array.isRequired,
+  container: PropTypes.object.isRequired,
+
+  scrollingElement: PropTypes.any,
+  options: PropTypes.object,
+  eventsListener: PropTypes.function,
+  totalItemsCount: PropTypes.number,
+  resizeMediaUrl: PropTypes.function,
+};
+
+export const LAYOUT_PROPS = {
+  ORIENTATION: 'orientation',
+  RTL: 'rtl',
+  SPACING: 'spacing',
+  SCROLL_DIRECTION: 'scrollDirection',
+  ITEM_SIZE: 'itemSize',
+  INFO: 'info',
+  PLAY: 'play',
+  OVERLAY_ANIMATION: 'overlayAnimation',
+  IMAGE_OVER_ANIMATION: 'imageHoverAnimation',
+  SCROLL_ANIMATION: 'scrollAnimation',
+  HOVERING_BEHAVIOUR: 'hoveringBehaviour',
+  LOOP: 'loop',
+  AUTO_SLIDE: 'autoSlide',
+  ITEMS_PER_COL: 'itemsPerCol',
+}
+
+export const baseLayoutPropTypes = {
+  orientaion: PropTypes.oneOf(['rows', 'cols']),
+  rtl: PropTypes.bool,
+  spacing: PropTypes.number,
+  scrollDirection: PropTypes.oneOf(['vertical', 'horizontal']),
+  itemSize: PropTypes.number,
+  info: PropTypes.oneOf(['above', 'below', 'hover', 'none']),
+  play: PropTypes.oneOf(['auto', 'hover', 'onClick']),
+  overlayAnimation: PropTypes.oneOf(consts.overlayAnimations),
+  imageHoverAnimation: PropTypes.oneOf(consts.imageHoverAnimations),
+  scrollAnimation: PropTypes.oneOf(consts.scrollAnimations),
+  hoveringBehaviour: PropTypes.oneOf(consts.hoveringBehaviours),
+  // cropType,
+  // cropRatio,
+  loop: PropTypes.bool, //slideshowLoop
+  autoSlide: PropTypes.number, //isAutoSlideshow && autoSlideshowInterval
+  itemsPerCol: PropTypes.number
+}
+
+export const layoutPropTypes = layoutProps => {
+  return {
+    ...basePropTypes,
+    ...layoutProps.reduce((resPropTypes, prop) => resPropTypes[prop] = baseLayoutPropTypes[prop], {})
+  }
+}
+
+export const propsToStyles = props => {
+  return {
+    ...props.styles,
```

Comment:
Move to bottom. protect from unwanted use. if users called something by name it will be the strongest.